### PR TITLE
feat: add Prometheus health check

### PR DESCRIPTION
**Pull request by roelvdberg**
PR implements healths check for monitoring.coreos.com/Prometheus

### CODE REVIEW

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          drop:
+            - ALL
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+  enableAdminAPI: false
+  evaluationInterval: 30s
+  externalUrl: 'http://prometheus-stack-kube-prom-prometheus.prometheus:9090'
+  image: 'quay.io/prometheus/prometheus:v2.37.0'
+  imagePullSecrets:
+    - name: mcps-registry-image-pull-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml:57]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        capabilities:
+          drop:
+            - ALL
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+  listenLocal: false
+  logFormat: logfmt
+  logLevel: info
+  nodeSelector:
+    ismcps.nl/workload: mcps
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml:71]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  resources:
+    requests:
+      memory: 700Mi
+  retention: 6h
+  routePrefix: /
+  ruleNamespaceSelector: {}
+  ruleSelector: {}
+  scrapeInterval: 10s
+  secrets:
+    - prometheus-federation-client-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml:88]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
...torSelector":{},"portName":"http-web","probeNamespaceSelector":{},"probeSelector":{},"replicas":2,"resources":{"requests":{"memory":"700Mi"}},"retention":"6h","routePrefix":"/","ruleNamespaceSelector":{},"ruleSelector":{},"scrapeInterval":"10s","secrets":["prometheus-federation-client-secret"],"securityContext":{"fsGroup":2000,"runAsGroup":2000,"runAsNonRoot":true,"runAsUser":1000},"serviceAccountName":"prometheus-stack-kube-prom-prometheus","serviceMonitorNamespaceSelector":{},"serviceMonitorSelector":{},"shards":1,"storage":{"volumeClaimTemplate":{"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"100Gi"}},"storageClassName":"default"}}},"thanos":{"objectStorageConfig":{"key":"config-file.yaml","name":"thanos-secret"}},"tolerations":[{"key":"CriticalAddonsOnly","operator":"Exists"}],"topologySpreadConstraints":[{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"kubernetes.io/hostname","whenUnsatisfiable":"ScheduleAnyway"},{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"topology.kubernetes.io/zone","whenUnsatisfiable":"DoNotSchedule"}],"version":"v2.37.0"}}
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml:8]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+          requests:
+            storage: 100Gi
+        storageClassName: default
+  thanos:
+    objectStorageConfig:
+      key: config-file.yaml
+      name: thanos-secret
+  tolerations:
+    - key: CriticalAddonsOnly
+      operator: Exists
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/degraded.yaml:113]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
...torSelector":{},"portName":"http-web","probeNamespaceSelector":{},"probeSelector":{},"replicas":2,"resources":{"requests":{"memory":"700Mi"}},"retention":"6h","routePrefix":"/","ruleNamespaceSelector":{},"ruleSelector":{},"scrapeInterval":"10s","secrets":["prometheus-federation-client-secret"],"securityContext":{"fsGroup":2000,"runAsGroup":2000,"runAsNonRoot":true,"runAsUser":1000},"serviceAccountName":"prometheus-stack-kube-prom-prometheus","serviceMonitorNamespaceSelector":{},"serviceMonitorSelector":{},"shards":1,"storage":{"volumeClaimTemplate":{"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"100Gi"}},"storageClassName":"default"}}},"thanos":{"objectStorageConfig":{"key":"config-file.yaml","name":"thanos-secret"}},"tolerations":[{"key":"CriticalAddonsOnly","operator":"Exists"}],"topologySpreadConstraints":[{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"kubernetes.io/hostname","whenUnsatisfiable":"ScheduleAnyway"},{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"topology.kubernetes.io/zone","whenUnsatisfiable":"DoNotSchedule"}],"version":"v2.37.0"}}
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml:7]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        drop:
+        - ALL
+      readOnlyRootFilesystem: true
+      runAsNonRoot: true
+  enableAdminAPI: false
+  evaluationInterval: 30s
+  externalUrl: http://prometheus-stack-kube-prom-prometheus.prometheus:9090
+  image: quay.io/prometheus/prometheus:v2.37.0
+  imagePullSecrets:
+  - name: mcps-registry-image-pull-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml:56]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+      capabilities:
+        drop:
+        - ALL
+      readOnlyRootFilesystem: true
+      runAsNonRoot: true
+  listenLocal: false
+  logFormat: logfmt
+  logLevel: info
+  nodeSelector:
+    ismcps.nl/workload: mcps
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml:70]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  resources:
+    requests:
+      memory: 700Mi
+  retention: 6h
+  routePrefix: /
+  ruleNamespaceSelector: {}
+  ruleSelector: {}
+  scrapeInterval: 10s
+  secrets:
+  - prometheus-federation-client-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml:87]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+          requests:
+            storage: 100Gi
+        storageClassName: default
+  thanos:
+    objectStorageConfig:
+      key: config-file.yaml
+      name: thanos-secret
+  tolerations:
+  - key: CriticalAddonsOnly
+    operator: Exists
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/healthy.yaml:112]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
...torSelector":{},"portName":"http-web","probeNamespaceSelector":{},"probeSelector":{},"replicas":2,"resources":{"requests":{"memory":"700Mi"}},"retention":"6h","routePrefix":"/","ruleNamespaceSelector":{},"ruleSelector":{},"scrapeInterval":"10s","secrets":["prometheus-federation-client-secret"],"securityContext":{"fsGroup":2000,"runAsGroup":2000,"runAsNonRoot":true,"runAsUser":1000},"serviceAccountName":"prometheus-stack-kube-prom-prometheus","serviceMonitorNamespaceSelector":{},"serviceMonitorSelector":{},"shards":1,"storage":{"volumeClaimTemplate":{"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"100Gi"}},"storageClassName":"default"}}},"thanos":{"objectStorageConfig":{"key":"config-file.yaml","name":"thanos-secret"}},"tolerations":[{"key":"CriticalAddonsOnly","operator":"Exists"}],"topologySpreadConstraints":[{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"kubernetes.io/hostname","whenUnsatisfiable":"ScheduleAnyway"},{"labelSelector":{"matchLabels":{"app.kubernetes.io/name":"prometheus"}},"maxSkew":1,"topologyKey":"topology.kubernetes.io/zone","whenUnsatisfiable":"DoNotSchedule"}],"version":"v2.37.0"}}
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml:8]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+          drop:
+            - ALL
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+  enableAdminAPI: false
+  evaluationInterval: 30s
+  externalUrl: 'http://prometheus-stack-kube-prom-prometheus.prometheus:9090'
+  image: 'quay.io/prometheus/prometheus:v2.37.0'
+  imagePullSecrets:
+    - name: mcps-registry-image-pull-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml:57]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+        capabilities:
+          drop:
+            - ALL
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+  listenLocal: false
+  logFormat: logfmt
+  logLevel: info
+  nodeSelector:
+    ismcps.nl/workload: mcps
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml:71]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+  resources:
+    requests:
+      memory: 700Mi
+  retention: 6h
+  routePrefix: /
+  ruleNamespaceSelector: {}
+  ruleSelector: {}
+  scrapeInterval: 10s
+  secrets:
+    - prometheus-federation-client-secret
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml:88]:
```suggestion
```

File: resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+          requests:
+            storage: 100Gi
+        storageClassName: default
+  thanos:
+    objectStorageConfig:
+      key: config-file.yaml
+      name: thanos-secret
+  tolerations:
+    - key: CriticalAddonsOnly
+      operator: Exists
```
Comment (rouke-broersma) [inline resource_customizations/monitoring.coreos.com/Prometheus/testdata/progressing.yaml:113]:
```suggestion
```

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks @roelvdberg!

**Pull request merged**